src/test/regress/expected/distributed_locks.out

```diff
@@ -0,0 +1,76 @@
+CREATE SCHEMA distribute_lock_tests;
+SET search_path TO distribute_lock_tests;
+SET citus.next_shard_id TO 10000;
+CREATE TABLE dist_table(a int);
+SELECT create_distributed_table('dist_table', 'a');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+INSERT INTO dist_table SELECT n FROM generate_series(1, 5) n;
+-- Test acquiring lock outside transaction
+LOCK dist_table IN ACCESS EXCLUSIVE MODE;
+ERROR:  LOCK TABLE can only be used in transaction blocks
+-- Test acquiring lock inside procedure
+DO $$
+BEGIN
+LOCK dist_table IN ACCESS EXCLUSIVE MODE;
+END;
+$$;
+-- Test that when the user does not have the required permissions to lock
+-- the locks are not forwarded to the workers
+CREATE USER read_only_user;
+NOTICE:  not propagating CREATE ROLE/USER commands to worker nodes
+HINT:  Connect to worker nodes directly to manually create all necessary users and roles.
+GRANT SELECT ON distribute_lock_tests.dist_table TO read_only_user;
+SET ROLE read_only_user;
+SET citus.log_remote_commands TO ON;
+BEGIN;
+LOCK distribute_lock_tests.dist_table IN ACCESS EXCLUSIVE MODE;
+ERROR:  permission denied for schema distribute_lock_tests
+ROLLBACK;
+RESET citus.log_local_commands;
+RESET ROLE;
+\c - - - :worker_1_port
+SET search_path TO distribute_lock_tests;
+-- Test trying to lock from a worker when the coordinator is not in the metadata
+SET citus.allow_unsafe_locks_from_workers TO 'off';
+BEGIN;
+LOCK dist_table IN ACCESS EXCLUSIVE MODE;
+ERROR:  Cannot acquire a distributed lock from a worker node since the coordinator is not in the metadata.
+HINT:  Either run this command on the coordinator or add the coordinator in the metadata by using: SELECT citus_set_coordinator_host('<hostname>', <port>);
+Alternatively, though it is not recommended, you can allow this command by running: SET citus.allow_unsafe_locks_from_workers TO 'on';
+ROLLBACK;
+-- Verify that the same restriction does not apply to worker local tables
+CREATE TABLE local_table(a int);
+-- Verify that no locks will be distributed for the local lock
+SET citus.log_remote_commands TO ON;
+BEGIN;
+LOCK local_table IN ACCESS EXCLUSIVE MODE;
+ROLLBACK;
+RESET citus.log_remote_commands;
+-- Cleanup local table
+DROP TABLE local_table;
+-- Test that setting the guc to 'on' will allow the lock from workers
+SET citus.allow_unsafe_locks_from_workers TO 'on';
+BEGIN;
+LOCK dist_table IN ACCESS EXCLUSIVE MODE;
```

Comment:
Two quick notes:

1) we don't seem to have any successful tests from the coordinator?
2) it would be worthwhile to use all the lock modes, just to make sure we properly deparse all the lock modes 
```SQL

-- Try all valid lock options; also try omitting the optional TABLE keyword.
BEGIN TRANSACTION;
LOCK TABLE dist_table IN ACCESS SHARE MODE;
LOCK dist_table IN ROW SHARE MODE;
LOCK TABLE dist_table IN ROW EXCLUSIVE MODE;
LOCK TABLE dist_table IN SHARE UPDATE EXCLUSIVE MODE;
LOCK TABLE dist_table IN SHARE MODE;
LOCK dist_table IN SHARE ROW EXCLUSIVE MODE;
LOCK TABLE dist_table IN EXCLUSIVE MODE;
LOCK TABLE dist_table IN ACCESS EXCLUSIVE MODE;
ROLLBACK;
```